src/editor/operations.ts

```diff
@@ -205,8 +209,18 @@ export function formatRangeAsCode(range: Range): void {
             parts.push(partCreator.newline());
         }
     } else {
-        toggleInlineFormat(range, "`");
-        return;
+        const fenceLen = longestBacktickSequence(range.text);
+        if (hasBacktick) {
+            parts.unshift(partCreator.plain("`".repeat(fenceLen + 1)));
+            parts.push(partCreator.plain("`".repeat(fenceLen+ 1)));
+        } else {
+            if (fenceLen === 0) {
```

Comment:
Can we collapse the else-ifs?

Reply:
Ikr, Trust me, i tried many variations but it doesn't seem to work except i do it this way :(